core/src/bmi.jl

```diff
@@ -12,6 +22,14 @@ Write all results to the configured files.
 """
 function BMI.finalize(model::Model)::Nothing
     write_results(model)
+    log_finalize(model)
+    logger = global_logger()
+    try
+        io = logger.logger.loggers[1].logger.logger.stream
```

Comment:
Please rewrite this somehow, or explain why the rabbithole is so deep. Maybe also have a specific catch here?